R/db-odbc-mssql.R

```diff
@@ -218,4 +222,17 @@ mssql_generic_infix <- function(if_select, if_filter) {
   }
 }
 
+mssql_sql_if <- function(cond, if_true, if_false = NULL) {
+  build_sql(
+    "CASE WHEN ((", cond, ") =  'TRUE')", " THEN (", if_true, ")",
```

Comment:
Can you please check to see if this has the same problem as fixed in tidyverse/dplyr#3157

Also can you have a go at using `sql_expr()` here? It should make the code quite a bit simpler.